opm/common/OpmLog/LogBackend.cpp

```diff
@@ -27,18 +27,37 @@ namespace Opm {
     {
     }
 
+    LogBackend::~LogBackend()
+    {
+    }
 
-    bool LogBackend::includeMessage(int64_t messageFlag) {
-        if (((messageFlag & m_mask) == messageFlag) &&
-            (messageFlag > 0))
-            return true;
-        else
-            return false;
+    void LogBackend::configureDecoration(bool use_prefix, bool use_color_coding)
+    {
+        use_prefix_ = use_prefix;
+        use_color_coding_ = use_color_coding;
     }
 
-    int64_t LogBackend::getMask() const {
+    int64_t LogBackend::getMask() const
+    {
         return m_mask;
     }
 
+    bool LogBackend::includeMessage(int64_t messageFlag)
+    {
+        return ((messageFlag & m_mask) == messageFlag) && (messageFlag > 0);
+    }
+
+    std::string LogBackend::decorateMessage(int64_t messageFlag, const std::string& message)
```

Comment:
If the decoration process is based on a private method like:
```C++
std::string decorate_impl( const std::string& msg, const std::string& prefix , const std::string& suffix = "") {
     return prefix + msg + suffix;
}  
```
then both color coding and string based prefix / suffix can be implemented with the same code.